File: master/longest_common_prefix.py
class Solution:
    # @return a string
    def longestCommonPrefix(self, strs):
        if len(strs) == 0:
            return ""
        res = ""
        i = 1
        while True:
            if i>len(strs[0]):
                return res
            c = strs[0][i-1]
            for x in range(1,len(strs),1):
                if len(strs[x])<i or strs[x][i-1] != c:
                    return res
            res += c
            i += 1
        return res

if __name__ == "__main__":
    print Solution().longestCommonPrefix(["abcd", "abce", "abc", "ab"])

File: 6/longest_common_prefix.cpp
#include "header.h"

string longestCommonPrefix(vector<string>& strs) {
    int sz = strs.size();
    if(sz == 0) return "";
    if(sz == 1) return strs[0];
    string res = "";
    for(int i=0; i<strs[0].size(); i++){
        res.push_back(strs[0][i]);
        for(int j=1; j<sz; j++){
            if(i == strs[j].size() || strs[j][i] != strs[0][i]){
                res.pop_back();
                return res;
            }
        }
    }
    return res;
}

int main(){
//    string s0("abce");
//    string s1("abcd");
//    string s2("abde");
//    string s3("abcf");
//    vector<string> v;
//    v.push_back(s0);
//    v.push_back(s1);
//    v.push_back(s2);
//    v.push_back(s3);
    vector<string> v = {
        "ab",
        ""
    };
    string res = longestCommonPrefix(v);
    printf("%s\n", res.c_str());
    return 0;
}
